packages/logconsole-extension/src/index.tsx

```diff
@@ -118,6 +125,18 @@ function activateLogConsole(
     translator
   });
 
+  const getCurrentWidgetPath = () => {
+    const currentWidget = app.shell.currentWidget;
+    if (labShell?.currentPath) {
+      return labShell.currentPath;
+    }
+    // For other shells, set the source to the current widget path
+    if (currentWidget && currentWidget instanceof DocumentWidget) {
```

Comment:
For reference, this follows the same approach used by `ILabShell`:

https://github.com/jupyterlab/jupyterlab/blob/85c82eba1caa7e28a0d818c0840e13756c1b1256/packages/application/src/shell.ts#L1424-L1426